astropy/modeling/parameters.py

```diff
@@ -132,23 +116,29 @@ class Parameter(OrderedDescriptor):
     """
     Wraps individual parameters.
 
-    This class represents a model's parameter (in a somewhat broad sense).  It
-    acts as both a descriptor that can be assigned to a class attribute to
-    describe the parameters accepted by an individual model (this is called an
-    "unbound parameter"), or it can act as a proxy for the parameter values on
-    an individual model instance (called a "bound parameter").
+    Note: this is a new implementation of the Parameter class, and despite
+    the fact that it inherits from OrderedDescriptor, it no longer is a
+    descriptor.
+
+    This class represents a model's parameter (in a somewhat broad sense). It
+    serves a number of purposes:
+
+    1) A type to be recognized by models and treated specially at class
+    initialization (i.e., if it is found that there is a class definition
+    of a Parameter, the model initializer makes a copy at the instance level).
 
-    Parameter instances never store the actual value of the parameter directly.
-    Rather, each instance of a model stores its own parameters parameter values
-    in an array.  A *bound* Parameter simply wraps the value in a Parameter
-    proxy which provides some additional information about the parameter such
-    as its constraints.  In other words, this is a high-level interface to a
-    model's adjustable parameter values.
+    2) Managing the handling of allowable parameter values and once defined,
+    ensuring updates are consistent with the Parameter definition. This
+    includes the optional use of units and quantities as well as tranforming
+    values to an internally consistent representation (e.g., from degrees to
+    radians through the use of getters and setters).
 
-    *Unbound* Parameters are not associated with any specific model instance,
-    and are merely used by model classes to determine the names of their
-    parameters and other information about each parameter such as their default
-    values and default constraints.
+    3) Holding attributes of parameters relevant to fitting, such as whether
+    the parameter may be varied in fitting, or whether there are constraints
+    that must be satisfied.
+
+    Parameters now do store values locally (as instead previously in the
+    associated model)
```

Comment:
Add this to the section above that specifically discusses changes in 4.0

Reply:
done